addons/mail/tests/common.py

```diff
@@ -581,8 +581,20 @@ def assertSentEmail(self, author, recipients, **values):
                     email_to_list.append(formataddr((email_to.name, email_to.email or 'False')))
                 else:
                     email_to_list.append(email_to)
-        expected['email_to'] = email_to_list
+        formatted_emails = []
+        for email_to in email_to_list:
+            emails_split = email_to.split(', ')
```

Comment:
`email_split_tuples` already split on `,`, why is it needed ?

Reply:
@std-odoo suppose one of the emails in `email_to_list` is `'"Employee 1" <@e1>, "Employee 1" <dato@company.com>'`. in this case `email_split_tuples` returns [] for the first email, because it can't detect wrong/incorrect emails (which sometimes makes tests fail, because `_find_sent_email` can't find sent email, see second broken test [here](https://runbot.odoo.com/runbot/build/61290454)).  So what i do here is: split the multi mails and then individually format them and then combine them once again.

i agree the code is ugly and feel like there can be edge cases where it fails, but i don't know better way to resolve this  issue

Reply:
@std-odoo suppose one of the emails in email_to_list is '"Employee 1" <@e1>, "Employee 1" <dato@company.com>'. in this case email_split_tuples returns [] for the first email, because it can't detect wrong/incorrect emails (which sometimes makes tests fail, because _find_sent_email can't find sent email, see second broken test [here](https://runbot.odoo.com/runbot/build/61290454)). So what i do here is: split the multi mails and then individually format them and then combine them once again.

i agree the code is ugly and feel like there can be edge cases where it fails, but i don't know better way to resolve this issue